job_creator.php

```diff
@@ -486,8 +497,15 @@ public function createJson(string $yml): string
         }
         $matrix = ['include' => []];
 
+        if ($composerInstall) {
+            $json = json_decode(file_get_contents($this->composerJsonPath));
+            if (isset($json->config->platform->php) && preg_match('#^[0-9\.]+$#', $json->config->platform->php)) {
+                $this->phpVersionOverride = $json->config->platform->php;
+            }
+        }
+
         if ($dynamicMatrix) {
-            $matrix = $this->buildDynamicMatrix($matrix, $run, $simpleMatrix);
+            $matrix = $this->buildDynamicMatrix($matrix, $run, $composerInstall, $simpleMatrix);
```

Comment:
Responding to https://github.com/silverstripe/gha-generate-matrix/pull/35#issuecomment-1299189542here so we have a proper comment thread (I hate that you can't reply to comments like that)

>> given that, to set `$simpleMatrix` to `$simpleMatrix || $composerInstall`
>
> That's just adding confusion for no benefit?. `$simpleMatrix` == the input variable simple_matrix and `$composerInstall` == the input variable composer_install
>
> All this seems to do is make to override an input variable with another one ... for no benefit?

It replaces two variables which are always being used the same way for a single variable you can use in their place. It reduces complexity of every condition that uses them, which helps to _reduce_ confusion. If you find it confusing to reuse the `$simpleMatrix` variable for this, lets declare a new variable instead: `$reducedMatrix = $simpleMatrix || $composerInstall;` (or come up with some other name that describes the way these variables are used in tandem).

Ultimately I won't hold this PR back if you decide this doesn't reduce the confusion for you - but I think it's a good idea so I'm pinging it back this last time to recommend it.

Reply:
We still need them to be different because of 

```php
if ($composerInstall) {
            $json = json_decode(file_get_contents($this->composerJsonPath));
            if (isset($json->config->platform->php) && preg_match('#^[0-9\.]+$#', $json->config->platform->php)) {
                $this->phpVersionOverride = $json->config->platform->php;
            }
        }
```
i.e. the php version will get locked to `config.platform.php` if present, though only for $composerInstall, not $simpleMatrix

The variables $composerInstall and $simpleMatrix consistently == their input variable in multiple methods. If we change $simpleMatrix to equal either one of two input variables in some methods but not others, then it's adding confusion. Again, for virtually no benefit? I'm really struggling to see how adding a third `$reducedMatrix` is somehow making things clearer.  

We may also wish to modify what $simpleMatrix creates in the future, independent of the matrix created by $composerInstall

Reply:
I'll chalk this up to creative differences then and resolve the conversation.